Assets/BossRoom/Scripts/Shared/Net/GameNetPortal.cs

```diff
@@ -91,6 +103,13 @@ void Start()
             //we expect NetworkManager will expose an event like this itself.
             NetManager.OnServerStarted += OnNetworkReady;
             NetManager.OnClientConnectedCallback += ClientNetworkReadyWrapper;
+
+            //we register these without knowing whether we're a client or host. This is because certain messages can be sent super-early,
+            //before we even get our ClientConnected event (if acting as a client). It should be harmless to have server handlers registered
+            //on the client, because (a) nobody will be sending us these messages and (b) even if they did, nobody is listening for those
+            //server message events on the client anyway. 
+            RegisterClientMessageHandlers();
+            RegisterServerMessageHandlers();

```

Comment:
please open ticket MLAPI side, this is a flow that shouldn't happen.

Reply:
Done, Issue 799. https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/issues/799